pandas/tests/io/json/test_pandas.py

```diff
@@ -813,6 +813,41 @@ def test_reconstruction_index(self):
         result = read_json(df.to_json())
         assert_frame_equal(result, df)
 
+    @pytest.mark.parametrize(
+        "date_format,delta,formatted",
+        [
+            ("iso", ["1D", "2D"], ["P1DT0H0M0S", "P2DT0H0M0S"]),
+            ("epoch", ["1D", "2D"], [86400000, 172800000]),
+        ],
+    )
+    def test_series_timedelta_to_json(self, date_format, delta, formatted):
+        # GH28156: to_json not correctly formatting Timedelta
+        s = Series([pd.Timedelta(d) for d in delta])
+        s.index = ["one", "two"]
+
+        result = json.loads(s.to_json(date_format=date_format))
+        expected = {k: v for k, v in zip(s.index, formatted)}
+
+        assert result == expected
```

Comment:
Hmm can you make the test compare against the actual expected output? Right now I'm not sure this actually tests the date format, i.e. it could still always write out in epoch and just infer that back on the way in and test would still pass

Reply:
Fixed.